Main.java

```diff
@@ -0,0 +1,154 @@
+package com.company;
+
+import java.io.*;
+import java.util.*;
+
+public class Main {
+
+    public static void main(String[] args) throws IOException {

```

Comment:
Well tested, but better split to separate method where each method test its own case